// @keepkey/keepkey-sdk/dist/generated/models/MayachainSignAminoTransferRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * keepkey-sdk-server
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.1.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MayachainSignAminoTransferRequestToJSON = exports.MayachainSignAminoTransferRequestFromJSONTyped = exports.MayachainSignAminoTransferRequestFromJSON = exports.instanceOfMayachainSignAminoTransferRequest = void 0;
const TypesMayachainSignDocTransfer_1 = require("./TypesMayachainSignDocTransfer");
/**
 * Check if a given object implements the MayachainSignAminoTransferRequest interface.
 */
function instanceOfMayachainSignAminoTransferRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "signDoc" in value;
    isInstance = isInstance && "signerAddress" in value;
    return isInstance;
}
exports.instanceOfMayachainSignAminoTransferRequest = instanceOfMayachainSignAminoTransferRequest;
function MayachainSignAminoTransferRequestFromJSON(json) {
    return MayachainSignAminoTransferRequestFromJSONTyped(json, false);
}
exports.MayachainSignAminoTransferRequestFromJSON = MayachainSignAminoTransferRequestFromJSON;
function MayachainSignAminoTransferRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'signDoc': (0, TypesMayachainSignDocTransfer_1.TypesMayachainSignDocTransferFromJSON)(json['signDoc']),
        'signerAddress': json['signerAddress'],
    };
}
exports.MayachainSignAminoTransferRequestFromJSONTyped = MayachainSignAminoTransferRequestFromJSONTyped;
function MayachainSignAminoTransferRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'signDoc': (0, TypesMayachainSignDocTransfer_1.TypesMayachainSignDocTransferToJSON)(value.signDoc),
        'signerAddress': value.signerAddress,
    };
}
exports.MayachainSignAminoTransferRequestToJSON = MayachainSignAminoTransferRequestToJSON;
