// @keepkey/keepkey-sdk/dist/generated/models/MayachainSignAminoDepositRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * keepkey-sdk-server
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.1.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MayachainSignAminoDepositRequestToJSON = exports.MayachainSignAminoDepositRequestFromJSONTyped = exports.MayachainSignAminoDepositRequestFromJSON = exports.instanceOfMayachainSignAminoDepositRequest = void 0;
const TypesMayachainSignDocDeposit_1 = require("./TypesMayachainSignDocDeposit");
/**
 * Check if a given object implements the MayachainSignAminoDepositRequest interface.
 */
function instanceOfMayachainSignAminoDepositRequest(value) {
    let isInstance = true;
    isInstance = isInstance && "signDoc" in value;
    isInstance = isInstance && "signerAddress" in value;
    return isInstance;
}
exports.instanceOfMayachainSignAminoDepositRequest = instanceOfMayachainSignAminoDepositRequest;
function MayachainSignAminoDepositRequestFromJSON(json) {
    return MayachainSignAminoDepositRequestFromJSONTyped(json, false);
}
exports.MayachainSignAminoDepositRequestFromJSON = MayachainSignAminoDepositRequestFromJSON;
function MayachainSignAminoDepositRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'signDoc': (0, TypesMayachainSignDocDeposit_1.TypesMayachainSignDocDepositFromJSON)(json['signDoc']),
        'signerAddress': json['signerAddress'],
    };
}
exports.MayachainSignAminoDepositRequestFromJSONTyped = MayachainSignAminoDepositRequestFromJSONTyped;
function MayachainSignAminoDepositRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'signDoc': (0, TypesMayachainSignDocDeposit_1.TypesMayachainSignDocDepositToJSON)(value.signDoc),
        'signerAddress': value.signerAddress,
    };
}
exports.MayachainSignAminoDepositRequestToJSON = MayachainSignAminoDepositRequestToJSON;
