// @keepkey/keepkey-sdk/dist/generated/models/StdTxTransferMaya.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * keepkey-sdk-server
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.1.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StdTxTransferMayaToJSON = exports.StdTxTransferMayaFromJSONTyped = exports.StdTxTransferMayaFromJSON = exports.instanceOfStdTxTransferMaya = exports.StdTxTransferMayaTypeEnum = void 0;
const StdTxTransferValue_1 = require("./StdTxTransferValue");
/**
 * @export
 */
exports.StdTxTransferMayaTypeEnum = {
    mayachain_MsgSend: 'mayachain/MsgSend'
};
/**
 * Check if a given object implements the StdTxTransferMaya interface.
 */
function instanceOfStdTxTransferMaya(value) {
    let isInstance = true;
    isInstance = isInstance && "value" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
exports.instanceOfStdTxTransferMaya = instanceOfStdTxTransferMaya;
function StdTxTransferMayaFromJSON(json) {
    return StdTxTransferMayaFromJSONTyped(json, false);
}
exports.StdTxTransferMayaFromJSON = StdTxTransferMayaFromJSON;
function StdTxTransferMayaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'value': (0, StdTxTransferValue_1.StdTxTransferValueFromJSON)(json['value']),
        'type': json['type'],
    };
}
exports.StdTxTransferMayaFromJSONTyped = StdTxTransferMayaFromJSONTyped;
function StdTxTransferMayaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'value': (0, StdTxTransferValue_1.StdTxTransferValueToJSON)(value.value),
        'type': value.type,
    };
}
exports.StdTxTransferMayaToJSON = StdTxTransferMayaToJSON;
