// @keepkey/keepkey-sdk/dist/generated/models/StdTxDepositMaya.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * keepkey-sdk-server
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.1.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StdTxDepositMayaToJSON = exports.StdTxDepositMayaFromJSONTyped = exports.StdTxDepositMayaFromJSON = exports.instanceOfStdTxDepositMaya = exports.StdTxDepositMayaTypeEnum = void 0;
const StdTxDepositValue_1 = require("./StdTxDepositValue");
/**
 * @export
 */
exports.StdTxDepositMayaTypeEnum = {
    mayachain_MsgDeposit: 'mayachain/MsgDeposit'
};
/**
 * Check if a given object implements the StdTxDepositMaya interface.
 */
function instanceOfStdTxDepositMaya(value) {
    let isInstance = true;
    isInstance = isInstance && "value" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
exports.instanceOfStdTxDepositMaya = instanceOfStdTxDepositMaya;
function StdTxDepositMayaFromJSON(json) {
    return StdTxDepositMayaFromJSONTyped(json, false);
}
exports.StdTxDepositMayaFromJSON = StdTxDepositMayaFromJSON;
function StdTxDepositMayaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'value': (0, StdTxDepositValue_1.StdTxDepositValueFromJSON)(json['value']),
        'type': json['type'],
    };
}
exports.StdTxDepositMayaFromJSONTyped = StdTxDepositMayaFromJSONTyped;
function StdTxDepositMayaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'value': (0, StdTxDepositValue_1.StdTxDepositValueToJSON)(value.value),
        'type': value.type,
    };
}
exports.StdTxDepositMayaToJSON = StdTxDepositMayaToJSON;
