// @keepkey/keepkey-sdk/dist/generated/models/TypesMayachainSignDocTransfer.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * keepkey-sdk-server
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.1.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TypesMayachainSignDocTransferToJSON = exports.TypesMayachainSignDocTransferFromJSONTyped = exports.TypesMayachainSignDocTransferFromJSON = exports.instanceOfTypesMayachainSignDocTransfer = void 0;
const StdTxTransferMaya_1 = require("./StdTxTransferMaya");
const TypesThorchainSignDocTransferFee_1 = require("./TypesThorchainSignDocTransferFee");
/**
 * Check if a given object implements the TypesMayachainSignDocTransfer interface.
 */
function instanceOfTypesMayachainSignDocTransfer(value) {
    let isInstance = true;
    isInstance = isInstance && "fee" in value;
    isInstance = isInstance && "msgs" in value;
    isInstance = isInstance && "memo" in value;
    isInstance = isInstance && "sequence" in value;
    isInstance = isInstance && "chain_id" in value;
    isInstance = isInstance && "account_number" in value;
    return isInstance;
}
exports.instanceOfTypesMayachainSignDocTransfer = instanceOfTypesMayachainSignDocTransfer;
function TypesMayachainSignDocTransferFromJSON(json) {
    return TypesMayachainSignDocTransferFromJSONTyped(json, false);
}
exports.TypesMayachainSignDocTransferFromJSON = TypesMayachainSignDocTransferFromJSON;
function TypesMayachainSignDocTransferFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'fee': (0, TypesThorchainSignDocTransferFee_1.TypesThorchainSignDocTransferFeeFromJSON)(json['fee']),
        'msgs': (json['msgs'].map(StdTxTransferMaya_1.StdTxTransferMayaFromJSON)),
        'memo': json['memo'],
        'sequence': json['sequence'],
        'chain_id': json['chain_id'],
        'account_number': json['account_number'],
    };
}
exports.TypesMayachainSignDocTransferFromJSONTyped = TypesMayachainSignDocTransferFromJSONTyped;
function TypesMayachainSignDocTransferToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'fee': (0, TypesThorchainSignDocTransferFee_1.TypesThorchainSignDocTransferFeeToJSON)(value.fee),
        'msgs': (value.msgs.map(StdTxTransferMaya_1.StdTxTransferMayaToJSON)),
        'memo': value.memo,
        'sequence': value.sequence,
        'chain_id': value.chain_id,
        'account_number': value.account_number,
    };
}
exports.TypesMayachainSignDocTransferToJSON = TypesMayachainSignDocTransferToJSON;
